### DISCUSSION TITLE
GraphQL and Cypher

### DISCUSSION THREAD

**Discussion by jelflein** [orientechnologies/orientdb#9686] [Ideas] [answered]
Hey,

I think GraphQL support could be a good idea, because a lot of developers use this for graph databases. The other thing is, that you can implement. rest endpoints very good with GraphQL

https://graphql.org

A cypher support can be a good idea for transfer from Neo4J to OrientDB, because you can reuse the queries.

https://neo4j.com/developer/cypher/

By the way, OrientDB is an awesome DB, but the documentation is not very good, please write java doc and improve this.

**Comment by andrii0lomakin** [comment]
Hi @jelflein , we already have plans to improve documentation.  I suppose you will see changes in the upcoming months. As for GraphQL I personally think that it will be good to consider adding it. But as for cypher, I do not want to convert OrientDB into Neo4j, the latest one is purely a graph database. But we definitely need to consider adding migration support for the users.

**Reply by jelflein** [reply]
With GraphQL you can easily change the db server. For example you want easily switch from Neo4J to OrientDB, because you have the "queries" with GraphQL.

**Reply by dieeisenefaust** [reply]
Any update on GraphQL support?

**Comment by lvca** [comment, marked as answer]
Nobody will support GraphQL or Cypher for OrientDB, simply because OrientDB is a dead project ( :-( ), there is only a little maintenance activity by SAP. If you're interested in both GraphQL and Cypher support, please look at ArcadeDB as the natural evolution of OrientDB (https://blog.arcadedb.com/welcome-to-arcadedb):
- GraphQL: https://docs.arcadedb.com/#GraphQL
- Cypher: https://docs.arcadedb.com/#Open-Cypher

**Reply by tglman** [reply]
I'm happy to see also other project working on this, even though this looks more like a marketing pitch than an answer, but if any user want to try the new shiny thing, here it is go and enjoy it !

**Comment by tglman** [comment]
Hi,

As today there is no plan for direct support for either graphql or open cypher, if some quality work will be done by third party, we will be happy to review and merge, but probably make more sense to start to think to a common standard like: https://www.gqlstandards.org/ 

Bye
